### ISSUE TITLE
ANR related to com.google.firebase.iid.FirebaseInstanceIdReceiver

### ISSUE THREAD

**Issue by LogeshM-rupeek** [firebase/firebase-android-sdk#2860]
###  Environment

  * Android Studio version: 4.2.1
  * Firebase Component: Messaging
  * Component version: BoM v-28.2.0

### Problem
We are getting this ANR
Broadcast of Intent { act=com.google.android.c2dm.intent.RECEIVE flg=0x10 pkg=com.company.customer cmp=com.company.customer/com.google.firebase.iid.FirebaseInstanceIdReceiver (has extras) }

on updating the existing individual dependency from

   api 'com.google.firebase:firebase-perf:19.1.1'
    api 'com.google.firebase:firebase-core:18.0.3'
    api 'com.google.firebase:firebase-messaging:21.1.0'
    api 'com.google.firebase:firebase-config:20.0.4'
    api 'com.google.firebase:firebase-analytics:18.0.3'
    api 'com.google.firebase:firebase-crashlytics:17.4.1'

to BoM version of it
  implementation platform('com.google.firebase:firebase-bom:28.2.0')
    api 'com.google.firebase:firebase-perf'
    api 'com.google.firebase:firebase-messaging'
    api 'com.google.firebase:firebase-config'
    api 'com.google.firebase:firebase-analytics'
    api 'com.google.firebase:firebase-crashlytics'

our target and min sdk is,
targetSdkVersion 29
minSdkVersion 21

In Google play console there is no stack trace on this crash. Please let us know about the update because we are facing this crash only after updating to BoM version of firebase.

Thanks

#### Steps to reproduce:
Occurring due to change of firebase individual SDK versions to Firebase BoM v-28.2.0

#### Relevant Code:

```
implementation platform('com.google.firebase:firebase-bom:28.2.0')
    api 'com.google.firebase:firebase-perf'
    api 'com.google.firebase:firebase-messaging'
    api 'com.google.firebase:firebase-config'
    api 'com.google.firebase:firebase-analytics'
    api 'com.google.firebase:firebase-crashlytics'

```

**Comment by gsakakihara** [comment]
Unfortunately that is not really enough information to determine where the problem is, but a few questions that might help narrow down the issue:

1) Do the bug reports include any additional information, like the stack traces/status of the threads for the device when the ANR occurred?
2) Did you see any ANRs on any previous versions?
3) Was updating the SDK versions the only change made between the version of the app that you started seeing this and the previous version?
4) Do you use the FirebaseMessaging methods for getting FCM tokens or do you use FirebaseInstanceId methods?

**Comment by LogeshM-rupeek** [comment]
1. As said in the issue there is no stack trace in google play console. The only information available is device information.

Panasonic P95 brightness_1 Android 7.1 (SDK 25)
 Nokia 6.1 brightness_1 Android 10 (SDK 29)
 Samsung Galaxy J7 brightness_1 Android 6.0 (SDK 23)
 Vivo vivo 1726 brightness_1 Android 8.1 (SDK 27)
 Vivo vivo 1902 brightness_1 Android 9 (SDK 28)
 Samsung Galaxy A50s brightness_1 Android 11 (SDK 30)
 Oppo realme 1 brightness_1 Android 9 (SDK 28)
These are the set of devices where the ANR occured.

2. Yes, but not this ANR. That is related to JobInfoSchedulerService. Issue reported in Issue tracker https://issuetracker.google.com/issues/191393213

3. Yes updating the SDK from individual versions to BoM caused this new ANR issue.

4.  Using Firebase messaging method only.

` FirebaseMessaging.getInstance().token
                .addOnCompleteListener(OnCompleteListener { task ->
                    if (!task.isSuccessful) {
                        Log.w(Constants.SHARED_PREF_KEY, "getInstanceId failed", task.exception)
                        return@OnCompleteListener
                    }})`

**Comment by gsakakihara** [comment]
If it's FCM related, there is not much of a difference between 21.1.0 and 22.0.0 (the version included in BoM v-28.2.0) other than the removal of the firebase-iid dependency. You could try adding that dependency back in to see if that's what's causing the ANR, otherwise it could also be one of the other Firebase components being updated, but I'm not familiar with what has changed between those versions.

**Comment by eygraber** [comment]
I'm getting this on 28.2.0:

```
Broadcast of Intent { act=com.google.android.c2dm.intent.RECEIVE flg=0x11000010 pkg=com.myapp cmp=com.myapp/com.google.firebase.iid.FirebaseInstanceIdReceiver (has extras) }
```

Main:
```
  #00  pc 0000000000019eec  /system/lib/libc.so (syscall+28)
  #00  pc 000000000001d23b  /system/lib/libc.so (__futex_wait_ex(void volatile*, bool, int, bool, timespec const*)+90)
  #00  pc 00000000000721bf  /system/lib/libc.so (pthread_cond_wait+32)
  #00  pc 000000000004b0c5  /system/lib/libc++.so (std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&)+8)
  #00  pc 000000000004cf8d  /system/lib/libc++.so (std::__1::__assoc_sub_state::copy()+56)
  #00  pc 000000000004d26b  /system/lib/libc++.so (std::__1::future<void>::get()+10)
  #00  pc 000000000008d4f3  /system/lib/libhwui.so (android::uirenderer::renderthread::RenderProxy::setStopped(bool)+190)
  at android.view.ThreadedRenderer.nSetStopped (Native method)
  at android.view.ThreadedRenderer.setStopped (ThreadedRenderer.java:524)
  at android.view.ViewRootImpl.performDraw (ViewRootImpl.java:3329)
  at android.view.ViewRootImpl.performTraversals (ViewRootImpl.java:2652)
  at android.view.ViewRootImpl.doTraversal (ViewRootImpl.java:1571)
  at android.view.ViewRootImpl$TraversalRunnable.run (ViewRootImpl.java:7615)
  at android.view.Choreographer$CallbackRecord.run (Choreographer.java:1034)
  at android.view.Choreographer.doCallbacks (Choreographer.java:845)
  at android.view.Choreographer.doFrame (Choreographer.java:780)
  at android.view.Choreographer$FrameDisplayEventReceiver.run (Choreographer.java:1020)
  at android.os.Handler.handleCallback (Handler.java:873)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at android.os.Looper.loop (Looper.java:205)
  at android.app.ActivityThread.main (ActivityThread.java:6991)
  at java.lang.reflect.Method.invoke (Native method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:493)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:884)
```

Jit thread pool worker thread 0:
```
#00  pc 00000000002d9df3  /system/lib/libart.so (art::DumpNativeStack(std::__1::basic_ostream<char, std::__1::char_traits<char>>&, int, BacktraceMap*, char const*, art::ArtMethod*, void*, bool)+134)
  #00  pc 000000000036f103  /system/lib/libart.so (art::Thread::DumpStack(std::__1::basic_ostream<char, std::__1::char_traits<char>>&, bool, BacktraceMap*, bool) const+210)
  #00  pc 000000000036b8bb  /system/lib/libart.so (art::Thread::Dump(std::__1::basic_ostream<char, std::__1::char_traits<char>>&, bool, BacktraceMap*, bool) const+34)
  #00  pc 0000000000384575  /system/lib/libart.so (art::DumpCheckpoint::Run(art::Thread*)+624)
  #00  pc 000000000036fae3  /system/lib/libart.so (art::Thread::RunCheckpointFunction()+142)
  #00  pc 000000000024c29f  /system/lib/libart.so (art::jit::JitCompileTask::Run(art::Thread*)+530)
  #00  pc 0000000000384ff1  /system/lib/libart.so (art::ThreadPoolWorker::Run()+44)
  #00  pc 0000000000384c37  /system/lib/libart.so (art::ThreadPoolWorker::Callback(void*)+94)
  #00  pc 00000000000729d5  /system/lib/libc.so (__pthread_start(void*)+22)
  #00  pc 000000000001dec5  /system/lib/libc.so (__start_thread+24)
```

Binder:21257_1:
```
 #00  pc 00000000000612f8  /system/lib/libc.so (__ioctl+8)
  #00  pc 0000000000022c73  /system/lib/libc.so (ioctl+30)
  #00  pc 000000000003eb73  /system/lib/libbinder.so (android::IPCThreadState::talkWithDriver(bool)+174)
  #00  pc 000000000003ecbd  /system/lib/libbinder.so (android::IPCThreadState::getAndExecuteCommand()+8)
  #00  pc 000000000003f21d  /system/lib/libbinder.so (android::IPCThreadState::joinThreadPool(bool)+40)
  #00  pc 0000000000056655  /system/lib/libbinder.so (android::PoolThread::threadLoop()+12)
  #00  pc 000000000000c6af  /system/lib/libutils.so (android::Thread::_threadLoop(void*)+202)
  #00  pc 00000000000724e9  /system/lib/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+88)
  #00  pc 00000000000729d5  /system/lib/libc.so (__pthread_start(void*)+22)
  #00  pc 000000000001dec5  /system/lib/libc.so (__start_thread+24)
```

Binder:21257_2:
```
 #00  pc 0000000000019eec  /system/lib/libc.so (syscall+28)
  #00  pc 00000000000a6b9f  /system/lib/libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*)+78)
  #00  pc 000000000026b57d  /system/lib/libart.so (art::JNI::CallBooleanMethodV(_JNIEnv*, _jobject*, _jmethodID*, std::__va_list)+344)
  #00  pc 000000000007d495  /system/lib/libandroid_runtime.so (_JNIEnv::CallBooleanMethod(_jobject*, _jmethodID*, ...)+28)
  #00  pc 00000000000cc591  /system/lib/libandroid_runtime.so (JavaBBinder::onTransact(unsigned int, android::Parcel const&, android::Parcel*, unsigned int)+96)
  #00  pc 0000000000036101  /system/lib/libbinder.so (android::BBinder::transact(unsigned int, android::Parcel const&, android::Parcel*, unsigned int)+72)
  #00  pc 000000000003ef6b  /system/lib/libbinder.so (android::IPCThreadState::executeCommand(int)+394)
  #00  pc 000000000003ed1b  /system/lib/libbinder.so (android::IPCThreadState::getAndExecuteCommand()+102)
  #00  pc 000000000003f23d  /system/lib/libbinder.so (android::IPCThreadState::joinThreadPool(bool)+72)
  #00  pc 0000000000056655  /system/lib/libbinder.so (android::PoolThread::threadLoop()+12)
  #00  pc 000000000000c6af  /system/lib/libutils.so (android::Thread::_threadLoop(void*)+202)
  #00  pc 00000000000724e9  /system/lib/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+88)
  #00  pc 00000000000729d5  /system/lib/libc.so (__pthread_start(void*)+22)
  #00  pc 000000000001dec5  /system/lib/libc.so (__start_thread+24)
```

Binder:21257_3:
```
#00  pc 0000000000019eec  /system/lib/libc.so (syscall+28)
  #00  pc 00000000000a6b9f  /system/lib/libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*)+78)
  #00  pc 000000000026b57d  /system/lib/libart.so (art::JNI::CallBooleanMethodV(_JNIEnv*, _jobject*, _jmethodID*, std::__va_list)+344)
  #00  pc 000000000007d495  /system/lib/libandroid_runtime.so (_JNIEnv::CallBooleanMethod(_jobject*, _jmethodID*, ...)+28)
  #00  pc 00000000000cc591  /system/lib/libandroid_runtime.so (JavaBBinder::onTransact(unsigned int, android::Parcel const&, android::Parcel*, unsigned int)+96)
  #00  pc 0000000000036101  /system/lib/libbinder.so (android::BBinder::transact(unsigned int, android::Parcel const&, android::Parcel*, unsigned int)+72)
  #00  pc 000000000003ef6b  /system/lib/libbinder.so (android::IPCThreadState::executeCommand(int)+394)
  #00  pc 000000000003ed1b  /system/lib/libbinder.so (android::IPCThreadState::getAndExecuteCommand()+102)
  #00  pc 000000000003f23d  /system/lib/libbinder.so (android::IPCThreadState::joinThreadPool(bool)+72)
  #00  pc 0000000000056655  /system/lib/libbinder.so (android::PoolThread::threadLoop()+12)
  #00  pc 000000000000c6af  /system/lib/libutils.so (android::Thread::_threadLoop(void*)+202)
  #00  pc 00000000000724e9  /system/lib/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+88)
  #00  pc 00000000000729d5  /system/lib/libc.so (__pthread_start(void*)+22)
  #00  pc 000000000001dec5  /system/lib/libc.so (__start_thread+24)
```

Profile Saver:
```
#00  pc 0000000000019ef0  /system/lib/libc.so (syscall+32)
  #00  pc 00000000000a6ecb  /system/lib/libart.so (art::ConditionVariable::TimedWait(art::Thread*, long long, int)+98)
  #00  pc 000000000025ddd1  /system/lib/libart.so (art::ProfileSaver::Run()+632)
  #00  pc 0000000000260591  /system/lib/libart.so (art::ProfileSaver::RunProfileSaverThread(void*)+52)
  #00  pc 00000000000729d5  /system/lib/libc.so (__pthread_start(void*)+22)
  #00  pc 000000000001dec5  /system/lib/libc.so (__start_thread+24)
```

LocationSettingsChangeThread:
```
#00  pc 0000000000019eec  /system/lib/libc.so (syscall+28)
  #00  pc 00000000000a6b9f  /system/lib/libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*)+78)
  #00  pc 00000000003d4c4f  /system/lib/libart.so (art::GoToRunnable(art::Thread*) (.llvm.1051389574)+314)
  #00  pc 00000000003d4aed  /system/lib/libart.so (art::JniMethodEnd(unsigned int, art::Thread*)+8)
  at android.os.MessageQueue.nativePollOnce (Native method)
  at android.os.MessageQueue.next (MessageQueue.java:326)
  at android.os.Looper.loop (Looper.java:170)
  at android.os.HandlerThread.run (HandlerThread.java:65)
```

queued-work-looper:
```
#00  pc 0000000000019eec  /system/lib/libc.so (syscall+28)
  #00  pc 00000000000a6b9f  /system/lib/libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*)+78)
  #00  pc 00000000003d4c4f  /system/lib/libart.so (art::GoToRunnable(art::Thread*) (.llvm.1051389574)+314)
  #00  pc 00000000003d4aed  /system/lib/libart.so (art::JniMethodEnd(unsigned int, art::Thread*)+8)
  at android.os.MessageQueue.nativePollOnce (Native method)
  at android.os.MessageQueue.next (MessageQueue.java:326)
  at android.os.Looper.loop (Looper.java:170)
  at android.os.HandlerThread.run (HandlerThread.java:65)
```

Firebase-Messaging-Trigger-Topics-Io:
```
at java.lang.Object.wait (Native method)
  at java.lang.Thread.parkFor$ (Thread.java:2137)
  at sun.misc.Unsafe.park (Unsafe.java:358)
  at java.util.concurrent.locks.LockSupport.parkNanos (LockSupport.java:230)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos (AbstractQueuedSynchronizer.java:2101)
  at java.util.concurrent.LinkedBlockingQueue.poll (LinkedBlockingQueue.java:467)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1091)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1152)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:641)
  at yF4.run (:2)
  at java.lang.Thread.run (Thread.java:764)
```
Firebase-Messaging-Init:
```
at java.lang.Object.wait (Native method)
  at java.lang.Thread.parkFor$ (Thread.java:2137)
  at sun.misc.Unsafe.park (Unsafe.java:358)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:190)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await (AbstractQueuedSynchronizer.java:2059)
  at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take (ScheduledThreadPoolExecutor.java:1120)
  at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take (ScheduledThreadPoolExecutor.java:849)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1092)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1152)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:641)
  at yF4.run (:2)
  at java.lang.Thread.run (Thread.java:764)
```

Firebase-Messaging-Topics-Io:
```
at java.lang.Object.wait (Native method)
  at java.lang.Thread.parkFor$ (Thread.java:2137)
  at sun.misc.Unsafe.park (Unsafe.java:358)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:190)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await (AbstractQueuedSynchronizer.java:2059)
  at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take (ScheduledThreadPoolExecutor.java:1120)
  at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take (ScheduledThreadPoolExecutor.java:849)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1092)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1152)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:641)
  at yF4.run (:2)
  at java.lang.Thread.run (Thread.java:764)
```

TAG:
```
at java.lang.Object.wait (Native method)
  at java.lang.Thread.parkFor$ (Thread.java:2137)
  at sun.misc.Unsafe.park (Unsafe.java:358)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:190)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt (AbstractQueuedSynchronizer.java:868)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly (AbstractQueuedSynchronizer.java:1021)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly (AbstractQueuedSynchronizer.java:1328)
  at java.util.concurrent.CountDownLatch.await (CountDownLatch.java:232)
  at dC3.a (:8)
  at com.google.firebase.messaging.FirebaseMessaging.a
  at com.google.firebase.messaging.g.b (:1)
  at com.google.firebase.messaging.g.run (:31)
  at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:458)
  at java.util.concurrent.FutureTask.run (FutureTask.java:266)
  at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run (ScheduledThreadPoolExecutor.java:301)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1167)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:641)
  at yF4.run (:2)
  at java.lang.Thread.run (Thread.java:764)
```

Crashlytics Exception Handler1:
```
at java.lang.Object.wait (Native method)
  at java.lang.Thread.parkFor$ (Thread.java:2137)
  at sun.misc.Unsafe.park (Unsafe.java:358)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:190)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await (AbstractQueuedSynchronizer.java:2059)
  at java.util.concurrent.LinkedBlockingQueue.take (LinkedBlockingQueue.java:442)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1092)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1152)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:641)
  at Vv0.a (:1)
  at pp.run (:2)
  at java.lang.Thread.run (Thread.java:764)
```

GmsDynamite:
```
  at java.lang.Object.wait (Native method)
  at cz.run (com.google.android.gms.dynamite_dynamiteloader@212621061@21.26.21 (100300-0):2)
```

com.google.firebase.crashlytics.startup1:
```
#00  pc 0000000000019eec  /system/lib/libc.so (syscall+28)
  #00  pc 00000000000a6b9f  /system/lib/libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*)+78)
  #00  pc 00000000002973b9  /system/lib/libart.so (art::JNI::NewObjectArray(_JNIEnv*, int, _jclass*, _jobject*)+380)
  #00  pc 000000000001c9cb  /system/lib/libjavacore.so (Linux_android_getaddrinfo(_JNIEnv*, _jobject*, _jstring*, _jobject*, int)+362)
  at libcore.io.Linux.android_getaddrinfo (Native method)
  at libcore.io.BlockGuardOs.android_getaddrinfo (BlockGuardOs.java:172)
  at java.net.Inet6AddressImpl.lookupHostByName (Inet6AddressImpl.java:137)
  at java.net.Inet6AddressImpl.lookupAllHostAddr (Inet6AddressImpl.java:105)
  at java.net.InetAddress.getAllByName (InetAddress.java:1154)
  at com.android.okhttp.Dns$1.lookup (Dns.java:39)
  at com.android.okhttp.internal.http.RouteSelector.resetNextInetSocketAddress (RouteSelector.java:175)
  at com.android.okhttp.internal.http.RouteSelector.nextProxy (RouteSelector.java:141)
  at com.android.okhttp.internal.http.RouteSelector.next (RouteSelector.java:83)
  at com.android.okhttp.internal.http.StreamAllocation.findConnection (StreamAllocation.java:174)
  at com.android.okhttp.internal.http.StreamAllocation.findHealthyConnection (StreamAllocation.java:126)
  at com.android.okhttp.internal.http.StreamAllocation.newStream (StreamAllocation.java:95)
  at com.android.okhttp.internal.http.HttpEngine.connect (HttpEngine.java:281)
  at com.android.okhttp.internal.http.HttpEngine.sendRequest (HttpEngine.java:224)
  at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute (HttpURLConnectionImpl.java:461)
  at com.android.okhttp.internal.huc.HttpURLConnectionImpl.connect (HttpURLConnectionImpl.java:127)
  at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.connect (DelegatingHttpsURLConnection.java:89)
  at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.connect (HttpsURLConnectionImpl.java:26)
  at V41.b (:8)
  at uW2.a (:11)
  at Uv.run (:40)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1167)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:641)
  at Vv0.a (:1)
  at pp.run (:2)
  at java.lang.Thread.run (Thread.java:764)
```

firebase-installations-executor-1:
```
 at java.lang.Object.wait (Native method)
  at java.lang.Thread.parkFor$ (Thread.java:2137)
  at sun.misc.Unsafe.park (Unsafe.java:358)
  at java.util.concurrent.locks.LockSupport.parkNanos (LockSupport.java:230)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos (AbstractQueuedSynchronizer.java:2101)
  at java.util.concurrent.LinkedBlockingQueue.poll (LinkedBlockingQueue.java:467)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1091)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1152)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:641)
  at java.lang.Thread.run (Thread.java:764)
```


firebase-installations-executor-2
```
  at java.lang.Object.wait (Native method)
  at java.lang.Thread.parkFor$ (Thread.java:2137)
  at sun.misc.Unsafe.park (Unsafe.java:358)
  at java.util.concurrent.locks.LockSupport.parkNanos (LockSupport.java:230)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos (AbstractQueuedSynchronizer.java:2101)
  at java.util.concurrent.LinkedBlockingQueue.poll (LinkedBlockingQueue.java:467)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1091)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1152)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:641)
  at java.lang.Thread.run (Thread.java:764)
```
